frontend/packages/console-app/src/components/user-preferences/theme/usePreferredTheme.ts

```diff
@@ -0,0 +1,11 @@
+import { Dispatch, SetStateAction } from 'react';
+import { useUserSettings } from '@console/shared';
+
+export const usePreferredTheme = (): [string, Dispatch<SetStateAction<string>>, boolean] => {
+  const [preferredTheme, setPreferredTheme, preferredThemeLoaded] = useUserSettings<string>(
+    'console.theme',
+    'light',
```

Comment:
The default should be the system default using the `prefers-color-scheme: dark` media query. We should make the UI responsive when this changes (and we can add a listener for when the media query changes).

The user setting should be like the language setting where the default comes from the system. The preference lets you override the system default.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/onchange

Reply:
We planned to add a system default option later once we fix the known dark theme issues. So, that we do not have a broken experience with the dark theme. cc: @jerolimov

Reply:
Yep, the idea was to the "System default" preference later.

@vikram-raj If you want, you can create a story to keep Sams feedback and link there.

Or we add this already now BUT still sets the default to Light until we addressed the most UI glitches. Wdyt?

Reply:
Created a story to track the system default option. That we will visit later.

Reply:
@jerolimov @vikram-raj If we're worried about this, why are we exposing the preference at all right now? A user could still go into preferences and select it.

It's very easy to test by going into dev tools and adding the class to the HTML element without ever exposing this to the user.

Reply:
One of the reasons to add theme switcher dropdown now is that. It will make it easy to test dark theme while developing and fixing dark theme issues. And I agree with you that it is easy to add a class to HTML and test the dark theme but doing it after every change as the app gets refreshed is quite painful.

Reply:
Could we add a temporary localStorage setting? It's probably a couple lines of code.

I feel like we should either expose the setting and do it right, or not expose it at all yet. If we add a setting and we're not ready to ship this in 4.11, we'd have to back out code either way. The recent in-flight PRs should address many of the issues, though.